### ISSUE TITLE
kafka-eagle 监控不到消费组的数据

### ISSUE THREAD

**Issue by julietxy** [smartloli/kafka-eagle#86]
在kafka-eagle 日志里记录
 [2018-06-26 00:00:00] KafkaOffsetGetter.Thread-58 - ERROR - Get consumer records has error, msg is Error reading field 'metadata': String length -16913 cannot be negative

**Comment by smartloli** [comment]
@julietxy 
Try to use ```kafka.eagle.sql.fix.error=false```.

**Comment by julietxy** [comment]
hi：
     the kafka's version is 0.10.2.1 the kafka-console-consumer.sh's configuration is"
if [  "x$KAFKA_OPTS" ]; then
 export KAFKA_OPTS="-Djava.security.auth.login.config=/opt/beh/core/kafka/jass/cb_admin_jaas.conf"
fi

if [ "x$KAFKA_HEAP_OPTS" = "x" ]; then
    export KAFKA_HEAP_OPTS="-Xmx512M"
fi

exec $(dirname $0)/kafka-run-class.sh kafka.tools.ConsoleConsumer "$@"
[root@lf319-rh2288-113 bin]# cat kafka-console-consumer.sh |grep -v  "#"

if [  "x$KAFKA_OPTS" ]; then
 export KAFKA_OPTS="-Djava.security.auth.login.config=/opt/beh/core/kafka/jass/cb_admin_jaas.conf"
fi

if [ "x$KAFKA_HEAP_OPTS" = "x" ]; then
    export KAFKA_HEAP_OPTS="-Xmx512M"
fi

exec $(dirname $0)/kafka-run-class.sh kafka.tools.ConsoleConsumer "$@""

**Comment by julietxy** [comment]
@smartloli 
  the kafka's version is 0.10.2.1 the kafka-console-consumer.sh's configuration is"
if [ "x$KAFKA_OPTS" ]; then
export KAFKA_OPTS="-Djava.security.auth.login.config=/opt/beh/core/kafka/jass/cb_admin_jaas.conf"
fi

if [ "x$KAFKA_HEAP_OPTS" = "x" ]; then
export KAFKA_HEAP_OPTS="-Xmx512M"
fi

exec $(dirname $0)/kafka-run-class.sh kafka.tools.ConsoleConsumer "$@"
[root@lf319-rh2288-113 bin]# cat kafka-console-consumer.sh |grep -v "#"

if [ "x$KAFKA_OPTS" ]; then
export KAFKA_OPTS="-Djava.security.auth.login.config=/opt/beh/core/kafka/jass/cb_admin_jaas.conf"
fi

if [ "x$KAFKA_HEAP_OPTS" = "x" ]; then
export KAFKA_HEAP_OPTS="-Xmx512M"
fi

exec $(dirname $0)/kafka-run-class.sh kafka.tools.ConsoleConsumer "$@""

**Comment by smartloli** [comment]
@julietxy The reason is ```kafka-console-consumer.sh``` script used old kafka consumer api, if you use this script & consumer kafka topic data, then you can modify follower property.
```
# kafka.eagle.offset.storage=kafka
```
or
```
kafka.eagle.offset.storage=zookeeper
```

**Comment by julietxy** [comment]
@smartloli：
               hi ，thank you for your answer，it works。

**Comment by julietxy** [comment]
hi：
      now i can see the cousumer group on the ui,but i can not see the offset is 0 all of this,how can i change this appearance?



tianxiaoyan@bonc.com.cn
 
发件人： smartloli
发送时间： 2018-07-05 10:07
收件人： smartloli/kafka-eagle
抄送： julietxy; Mention
主题： Re: [smartloli/kafka-eagle] kafka-eagle 监控不到消费组的数据 (#86)
Closed #86.
